test/Microsoft.AspNet.Razor.Runtime.Test/DefaultTagHelperContentTest.cs

```diff
@@ -0,0 +1,388 @@
+            var tagHelperContent = new DefaultTagHelperContent();
+            var copiedTagHelperContent = new DefaultTagHelperContent();
+            var text1 = "Hello";
+            var text2 = " World!";
+            var expected = text1 + text2;
+            tagHelperContent.Append(text1);
+            tagHelperContent.Append(text2);
+
+            // Act
+            copiedTagHelperContent.Append(tagHelperContent);
+
+            // Assert
+            Assert.Equal(expected, copiedTagHelperContent.GetContent());
+        }
+
+        [Fact]
+        public void CanIdentifyIfModified_Set()
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+
+            // Act
+            tagHelperContent.SetContent("Test");
+
+            // Assert
+            Assert.True(tagHelperContent.IsModified);
+        }
+
+        
+        [Fact]
+        public void CanIdentifyIfModified_Append()
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+
+            // Act
+            tagHelperContent.Append("Foo");
+
+            // Assert
+            Assert.True(tagHelperContent.IsModified);
+        }
+
+        [Fact]
+        public void CanIdentifyIfModified_Clear()
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+
+            // Act
+            tagHelperContent.Clear();
+
+            // Assert
+            Assert.True(tagHelperContent.IsModified);
+        }
+
+        [Fact]
+        public void IsWhiteSpace_True_AfterMultipleWhiteSpaceAdditions()
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+            var whitespaces = "  ";
+
+            // Act
+            tagHelperContent.SetContent(whitespaces);
+
+            // Assert
+            Assert.Equal(whitespaces, tagHelperContent.GetContent());
+            Assert.True(tagHelperContent.IsWhiteSpace);
+        }
+
+        [Theory]
+        [InlineData("")]
+        [InlineData(" ")]
+        [InlineData("\n")]
+        [InlineData("\t")]
+        [InlineData("\r")]
+        public void CanIdentifyWhiteSpace(string data)
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+
+            // Act
+            tagHelperContent.SetContent("  ");
+            tagHelperContent.Append(data);
+
+            // Assert
+            Assert.True(tagHelperContent.IsWhiteSpace);
+        }
+
+        [Fact]
+        public void CanIdentifyWhiteSpace_WithoutIgnoringStrings()
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+
+            // Act
+            tagHelperContent.SetContent("  ");
+            tagHelperContent.Append("Hello");
+
+            // Assert
+            Assert.False(tagHelperContent.IsWhiteSpace);
+        }
+
+        [Fact]
+        public void CanIdentifyIsEmpty()
```

Comment:
something more like `IsEmpty_InitiallyFalse` here, `IsEmpty_FalseAfterSetEmptyContent`, and so on